### ISSUE TITLE
bin should prioritize updating itself

### ISSUE THREAD

**Issue by skorokithakis** [marcosnils/bin#139]
I just tried to update a few packages and got "no files found in zip file". I don't know what that's about, but there's a `bin` update too, and it occurs to me that that might fix the problem. I think it would be best if `bin` prioritized updating itself first, so if there was a failure due to a bug that had already been fixed, I could rerun it and it'd work the second time, rather than running it N times and waiting for it to randomly be updated first.

**Comment by marcosnils** [comment]
> I think it would be best if `bin` prioritized updating itself first

Even if we update bin at first, we'd have to "reload" the program for the next updates to take place which would require the user to run the command twice. In that sense, it's the same thing as doing: `bin update bin; bin update`

I'm not convinced about an apparent benefit of updating bin first.

**Comment by skorokithakis** [comment]
It would, but far more users retry when something fails than think to update `bin` itself, and even fewer know how to do that (I didn't, until I read your comment).

**Comment by breml** [comment]
Maybe `bin` could detect this as a special case and the launch it self again (which would then start the updated version of bin).

**Comment by marcosnils** [comment]
> Maybe `bin` could detect this as a special case and the launch it self again (which would then start the updated version of bin).

Yes.. this is an option as well.. but I personally believe it's not worth it. I don't find myself in many situations where an outdated versions of bin are causing issues unless a bug happened which "shouldn't" be a common scenario.

I'm not against if someone wants to implement it though.
